### ISSUE TITLE
[LTC] Introduce a `MetricsReport` python binding and allow backend to add their report as string

### ISSUE THREAD

**Issue by JackCaoG** [pytorch/pytorch#79592]
### 🚀 The feature, motivation and pitch

`Metrics` and `Counters` classes were originally introduced in https://github.com/pytorch/xla/blob/master/third_party/xla_client/metrics.h and then got upstream to LTC in https://github.com/pytorch/pytorch/blob/master/torch/csrc/lazy/core/metrics.cpp.

As part of the LTC migration, pytorch/xla wants to adapt the upstream `Metrics` class but we run into an issue. Currently pytorch/xla `Metrics` class is part of the `xla_client` which will be built with tensorflow. `xla_client` can not take any dependency to pytorch. `ComputationClinet` in the `xla_client` also needs to use `Metrics` to record some debugging information. I think it will be very hard to get rid of the pytorch/xla version of the `Metrics`.

Given that `xla_client` complications, I think we can still keep 2 `metrics` classes. When `MetricReport`(which we will need to add a binding on the LTC side) is being called in the LTC, LTC will collects all of its counters and metrics. After that is done, LTC can call a backendInterface to get the backend specified `MetricReport` as string and append that to its report.

I think it is actually a good idea since user can easily tell which metrics/counters are backend specified and where to look.

To summary I think we should
1. Add a `MetricsReport` python binding in LTC
2. Add a `GetBackendMetricReport` method to the `BackendInterface`
3. Append `GetBackendMetricReport` at the end of the LTC `MetricsReport`

@Krovatkin @wconstab 

### Alternatives

have two API one only collect metrics reports for LTC and another one only for xla

**Comment by wconstab** [comment]
Yea, I think this is a good idea too.

It's probably simplest as @JackCaoG proposed.  But just wondering, is there any reason we'd rather have the backend API return an object instead of a string from the backend?  (e.g. let us merge the metrics and format them)?   I am not going to push for this, i'd be fine with the string-append approach too.

**Comment by JackCaoG** [comment]
I think it is better to return string instead of `metric object` from backend because it is very hard to return the same `metric` project from our runtime given the dependency issue. I don't think we need to merge metrics even they have the same name, we can use the namespace for counter.

**Comment by wconstab** [comment]
yea, sounds ok to me.  go for it.

**Comment by Krovatkin** [comment]
https://github.com/pytorch/pytorch/pull/79679 https://github.com/pytorch/pytorch/pull/79678
